service/start.go

```diff
@@ -119,6 +115,132 @@ func (svc *service) startNostr(ctx context.Context, encryptionKey string) error
 	return nil
 }
 
+type createAppSubscriber struct {
+	events.EventSubscriber
+	svc   *service
+	relay *nostr.Relay
+}
+
+func (s *createAppSubscriber) ConsumeEvent(ctx context.Context, event *events.Event, globalProperties map[string]interface{}) {
+	switch event.Event {
+	case "app_created":
+		properties, ok := event.Properties.(map[string]interface{})
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to cast event.Properties to map")
+			return
+		}
+		walletChildPubkey, ok := properties["walletChildPubkey"].(string)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to get app walletChildPubkey")
+			return
+		}
+		id, ok := properties["id"].(uint)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to get app id")
+			return
+		}
+		if walletChildPubkey != "" {
+			go func() {
+				appWalletPrivKey, err := s.svc.keys.GetBIP32ChildKey(uint32(id))
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to calculate app wallet priv key")
+				}
+				err = s.svc.startAppWalletSubscription(ctx, s.relay, walletChildPubkey, appWalletPrivKey)
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to subscribe to wallet")
+				}
+				// TODO what should we do?
+			}()
+		}
+	}
+}
+
+type deleteAppSubscriber struct {
```

Comment:
I am wondering if we even need to do this. The next time Alby Hub is started or even next time Alby Hub connects to the relay, this should automatically be cleaned up?

Reply:
I think it's more hygienic to unsubscribe that nostr req and delete the 13194 event from the relay.
Without that the relay will keep events and subscriptions for deleted Apps that are no longer used.

Reply:
Maybe some clients use that 13194 event to check if a NWC connection exists or/and can be used, and that gets broken in unpredictable ways.
Besides it will be inconsistent behavior if a request 23194 with command get_info returns invalid connection but the info event 13194 still returns info for a wallet key that doesn't exist anymore.